src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const triangleSmallerSides = [a, b, c]
+      .filter((num) => num !== Math.max(a, b, c))
+      .reduce((cur, next) => cur + next, 0);
+
+    this.shape = 'triangle';
+    this.color = color;
```

Comment:
This assign is redundant as you are using typescript access modifiers in the parameter list.